trl/trainer/online_dpo_trainer.py

```diff
@@ -421,23 +425,39 @@ def training_step(
             ref_logprobs = torch.take_along_dim(ref_all_logprobs, completion_ids.unsqueeze(-1), dim=2).squeeze(-1)
             del ref_output, ref_logits, ref_all_logprobs  # free memory
 
-            # Get the reward from the reward model
+        # Get the reward from the reward model or judge:
+        if self.judge is not None:
+            prompts = self.processing_class.batch_decode(
+                prompt_completion_ids[:num_examples, :context_length], skip_special_tokens=True
+            )
+            completions = self.processing_class.batch_decode(
+                prompt_completion_ids[:, context_length:], skip_special_tokens=True
+            )
+
+            ranks_of_first_completion = self.judge.judge(
+                prompts, [[completions[i], completions[i + num_examples]] for i in range(num_examples)]
+            )
+            # convert ranks to a True/False mask:
+            # when rank == 0, it means the first completion is the best
+            # when rank == 1, it means the second completion is the best
+            mask = torch.tensor([rank == 1 for rank in ranks_of_first_completion], device=prompt_completion_ids.device)
```

Comment:
Can you double-check it's not this instead? It might be an error here
```suggestion
            mask = torch.tensor([rank == 0 for rank in ranks_of_first_completion], device=prompt_completion_ids.device)
```

Reply:
yes checking...